JavaScriptSPA/authPopup.js

```diff
@@ -14,10 +14,23 @@ function signIn() {
         
     }).catch(function (error) {
       console.log(error);
+
+      // Error handling
+      if (error.errorMessage) {
+        // Check for forgot password error
+        // Learn more about AAD error codes at https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-aadsts-error-codes
+        if (error.errorMessage.indexOf("AADB2C90118") > -1) {
```

Comment:
We've been given guidance from the server team in teh past to avoid writing logic around error codes like this. @sameerag is there any other way to write this logic?

Reply:
This is the only way to handle this atm. B2C is working on moving the password reset policy under the sign-up/sign-in, so this will be handled on the server side...but work is not done yet. All our samples do this. This should be the only time we are actually doing something like this.

---
In reply to: [416754769](https://github.com/Azure-Samples/active-directory-b2c-javascript-msal-singlepageapp/pull/60#discussion_r416754769) [](ancestors = 416754769)

Reply:
@jennyf19 Sounds good, thanks for the clarification!

Reply:
of course! one of our horrible dev experiences we are working to fix...lol.

---
In reply to: [416756992](https://github.com/Azure-Samples/active-directory-b2c-javascript-msal-singlepageapp/pull/60#discussion_r416756992) [](ancestors = 416756992)

Reply:
Looks good. I am curious if x-frame-deny is not seen here.

Reply:
X-FRAME-OPTIONS errors only manifest as timeouts, unfortunately.